### PR TITLE
Improve Append Mode doc

### PR DESCRIPTION
**Pull request by mapedraza**
Comes from: 
https://github.com/telefonicaid/iotagent-ul/issues/216#issuecomment-1590810453

### CODE REVIEW

File: doc/api.md [hunk 1]
```diff
@@ -448,6 +448,19 @@ This is a flag that can be enabled by activating the parameter `appendMode` in t
 activated, the update requests to the Context Broker will be performed always with APPEND type, instead of the default
 UPDATE. This have implications in the use of attributes with Context Providers, so this flag should be used with care.
 
+When running the agent using `appendMode=false`, if sending measures that are not included in the config group (as 
+active measures), the IoT Agent should return a `422 Unprocessable Entity` code with the following message:
```
Comment (fgalan) [inline doc/api.md:452]:
```suggestion
active measures), the IoT Agent returns a `422 Unprocessable Entity` code with the following message:
```

As I think is a description of functional behaviour and optionality is not possible here (I could be wrong).

File: doc/api.md [hunk 2]
```diff
@@ -448,6 +448,19 @@ This is a flag that can be enabled by activating the parameter `appendMode` in t
...
+
+```json
+{
+    "name": "ENTITY_GENERIC_ERROR",
+    "message": "Error accesing entity data for device: deviceType:dev of type: deviceType"
+}
+```
+
+Additionaly, The agent should have created the device and the corresponding entity in the broker if `autoprovision==true`
```
Comment (fgalan) [inline doc/api.md:461]:
```suggestion
Additionaly, the agent creates the device and the corresponding entity in the broker if `autoprovision==true` 
```

File: doc/api.md [hunk 3]
```diff
@@ -448,6 +448,19 @@ This is a flag that can be enabled by activating the parameter `appendMode` in t
...
+active measures), the IoT Agent returns a `422 Unprocessable Entity` code with the following message:
+
+```json
+{
+    "name": "ENTITY_GENERIC_ERROR",
+    "message": "Error accesing entity data for device: deviceType:dev of type: deviceType"
+}
+```
+
+Additionaly, the agent creates the device and the corresponding entity in the broker if `autoprovision==true`
```
Comment (mapedraza) [inline doc/api.md:461]:
```suggestion
Additionally, the agent creates the device and the corresponding entity in the broker if `autoprovision==true` 
```

**Comment by fgalan** [comment]
@Keshav-NEC it would be great if you could have a look to this PR and tell us if it solves https://github.com/telefonicaid/iotagent-ul/issues/216#issuecomment-1590810453

**Comment by Keshav-NEC** [comment]
Hi @fgalan, it would be great if someone will share the file path where I have to do changes in append mode.

**Comment by AlvaroVega** [comment]
> it would be great if someone will share the file path where I have to do changes in append mode.

You just need to define in docker env var   - IOTA_APPEND_MODE=true of in config.json add  key config: appendMode = true

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
Based in @AlvaroVega 's LGTM, this PR is going to be closed. @Keshav-NEC of course if new feedback comes after your test a new PR could be raised to improve documentation.

**Comment by AlvaroVega** [comment]
@mapedraza @fgalan My question here is if is really needed the option about appendMode = false, when we need it ? Why don't assume always is appendMode true an simplify everything?

**Comment by mapedraza** [comment]
The idea behind having appendMode=false is to not allow the IoT Agent creates new attributes on entities on the context broker side (similar to autoprovision but not the same) . I do not know if there is a practical scenario in which this is interesting. Probably is a good idea to change the default behavior to appendMode=true, at least, until further changes.

**Comment by fgalan** [comment]
Maybe related discussion: https://github.com/telefonicaid/iotagent-node-lib/pull/1111#issuecomment-1589279315

**Pull request merged**